### DISCUSSION TITLE
Implement $embed_dir to embed whole directory trees as maps

### DISCUSSION THREAD

**Discussion by trenta3** [vlang/v#11905] [Ideas]
Hi! $embed_file is very useful, but sometimes there are needs to include whole directories (like those that have lots of files), whose file names are known at compile time but cannot currently be included easily except by writing all of their paths explicitly.

Some kind of compile-time computation using `os.glob` would solve this, but I think it is much faster to currently include an $embed_dir mechanism.

**Comment by larpon** [comment]
I think such a thing will give too many headaches for the end user. V is usually very explicit and I think embedding by e.g. glob goes against this idea. You'll quickly loose track or forget what's being embedded.

An alternative I can think of is making a simple script that generates a list of files in `$embed_file('<file>')` code you can paste into your source instead. Also mind you that currently embedding will be slow on large data amounts (we hope to change this).
